barman/backup.py

```diff
@@ -611,6 +611,15 @@ def _validate_incremental_backup_configs(self):
                 "to perform an incremental backup using the Postgres backup method"
             )
 
+        parent_backup_id = kwargs.get("parent_backup_id")
+        parent_backup_info = self.get_backup(parent_backup_id)
+
+        if parent_backup_info and parent_backup_info.summarize_wal != "on":
+            raise BackupException(
+                "'summarize_wal' option has to be enabled in the Postgres server "
```

Comment:
The error here is different, the parent backup ID submitted is not valid because summarize_wal was not enabled when the backup has been executed, so the backup is not eligible for incremental.

The exception message should be changed

Reply:
@gcalacoci I rewrote the error message to be like this, what do you think?
"'summarize_wal' needs to be enabled in the Postgres Server for all backups in the backup chain for incremental mode. The summarize_wal field o the parent backup was not enabled in the Postgres server when it was taken."

Reply:
I'm not sure if it makes sense to mention a chain at this point, as a chain is more of a recovery thing.

Maybe we could have something like "The specified backup is not eligible as a parent for an incremental backup because WAL summaries were not enabled when that backup was taken." -- or something like that. Thoughts?

Reply:
agreed, sounds better!